test/Microsoft.AspNetCore.Mvc.TestCommon/CommonResourceInvokerTest.cs

```diff
@@ -1713,6 +1733,318 @@ public abstract class CommonResourceInvokerTest
+
+            var invoker = CreateInvoker(new IFilterMetadata[] { resourceFilter.Object, resultFilter.Object, });
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            resultFilter.Verify();
+        }
+
+        [Fact]
+        public async Task InvokeAction_ResourceFilterShortCircuit_DoesNotRunResultFilters()
+        {
+            // Arrange
+            var resultFilter1 = new Mock<IResultFilter>(MockBehavior.Strict);
+            resultFilter1.Setup(f => f.OnResultExecuting(It.IsAny<ResultExecutingContext>()));
+            resultFilter1.Setup(f => f.OnResultExecuted(It.IsAny<ResultExecutedContext>()));
+
+            var resultFilter2 = new Mock<IAsyncResultFilter>(MockBehavior.Strict);
+            resultFilter2.Setup(f => f.OnResultExecutionAsync(It.IsAny<ResultExecutingContext>(), It.IsAny<ResultExecutionDelegate>()));
+
+            var resourceFilter = new Mock<IResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecuting(It.IsAny<ResourceExecutingContext>()))
+                .Callback<ResourceExecutingContext>(c => c.Result = Result);
+
+            var invoker = CreateInvoker(new IFilterMetadata[] { resourceFilter.Object, resultFilter1.Object, resultFilter2.Object, });
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            resultFilter1.Verify(
+                f => f.OnResultExecuting(It.IsAny<ResultExecutingContext>()),
+                Times.Never());
+            resultFilter1.Verify(
+                f => f.OnResultExecuted(It.IsAny<ResultExecutedContext>()),
+                Times.Never());
+            resultFilter2.Verify(
+                f => f.OnResultExecutionAsync(It.IsAny<ResultExecutingContext>(), It.IsAny<ResultExecutionDelegate>()),
+                Times.Never());
+        }
+
+        [Fact]
+        public async Task InvokeAction_ExceptionFilterShortCircuit_InvokesAlwaysRunResultFilter()
+        {
+            // Arrange
+            var resultFilter = new Mock<IAlwaysRunResultFilter>(MockBehavior.Strict);
+            resultFilter.Setup(f => f.OnResultExecuting(It.IsAny<ResultExecutingContext>()))
+                .Callback<ResultExecutingContext>(c => Assert.Same(Result, c.Result))
+                .Verifiable();
+            resultFilter.Setup(f => f.OnResultExecuted(It.IsAny<ResultExecutedContext>()))
+                .Callback<ResultExecutedContext>(c => Assert.Same(Result, c.Result))
+                .Verifiable();
+
+            var exceptionFilter = new Mock<IExceptionFilter>(MockBehavior.Strict);
+            exceptionFilter
+                .Setup(f => f.OnException(It.IsAny<ExceptionContext>()))
+                .Callback<ExceptionContext>(c => c.Result = Result);
+
+            var invoker = CreateInvoker(new IFilterMetadata[] { exceptionFilter.Object, resultFilter.Object, }, Exception);
```

Comment:
Could you surround this IAlwaysRunResultFilter with regular IResultFilters (basically would be nice to seem them no-op). Applies to other similar tests too.

Reply:
I added that as part of tests that very we do not execute regular result filters.